recipes/cmake/3.x.x/conanfile.py

```diff
@@ -68,73 +70,73 @@ def validate(self):
+            if cross_building(self):
+                tc.variables["HAVE_POLL_FINE_EXITCODE"] = ''
+                tc.variables["HAVE_POLL_FINE_EXITCODE__TRYRUN_OUTPUT"] = ''
+            tc.generate()
 
     def build(self):
         if self.options.bootstrap:
-            with tools.chdir(self._source_subfolder):
-                self.run(['./bootstrap', '--prefix={}'.format(self.package_folder), '--parallel={}'.format(tools.cpu_count())])
-                autotools = AutoToolsBuildEnvironment(self)
+            toolchain_file_content = files.load_toolchain_args(self.generators_folder, namespace="bootstrap")
+            bootstrap_cmake_options = toolchain_file_content.get("bootstrap_cmake_options")
+            with chdir(self, self.source_folder):
+                self.run(f'./bootstrap --prefix="" --parallel={build_jobs(self)} {bootstrap_cmake_options}')
+                autotools = Autotools(self)
                 autotools.make()
         else:
-            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
-                                  "project(CMake)",
-                                  "project(CMake)\ninclude(\"{}/conanbuildinfo.cmake\")\nconan_basic_setup(NO_OUTPUT_DIRS)".format(
-                                      self.install_folder.replace("\\", "/")))
-            if self.settings.os == "Linux":
-                tools.replace_in_file(os.path.join(self._source_subfolder, "Utilities", "cmcurl", "CMakeLists.txt"),
-                                      "list(APPEND CURL_LIBS ${OPENSSL_LIBRARIES})",
-                                      "list(APPEND CURL_LIBS ${OPENSSL_LIBRARIES} ${CMAKE_DL_LIBS} pthread)")
-
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
+            cmake.configure()
             cmake.build()
 
     def package(self):
-        self.copy("Copyright.txt", dst="licenses", src=self._source_subfolder)
+        copy(self, "Copyright.txt", self.source_folder, os.path.join(self.package_folder, "licenses"), keep_path=False)
         if self.options.bootstrap:
-            with tools.chdir(self._source_subfolder):
-                autotools = AutoToolsBuildEnvironment(self)
+            with chdir(self, self.source_folder):
+                autotools = Autotools(self)
                 autotools.install()
         else:
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
             cmake.install()
         rmdir(self, os.path.join(self.package_folder, "doc"))
 
     def package_id(self):
         del self.info.settings.compiler
+        del self.info.options.bootstrap
 
     def package_info(self):
-        module_version = "{}.{}".format(Version(self.version).major, Version(self.version).minor)
-
-        bindir = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH environment variable: {}".format(bindir))
-        self.env_info.PATH.append(bindir)
```

Comment:
Do not remove please. Currently it works because test package is broken. It tests cmake as a requires, so bindirs are injected to `PATH` when you execute `self.run(..., run_environment=True)`, but a proper test v1 package would fail (cmake in build requirements).